src/Plover/Reduce.hs

```diff
@@ -290,10 +298,22 @@ typeCheck' (Vec var bound body) = do
 typeCheck' (Ref var) = varType var
 -- TODO get rid of this Expr?
 typeCheck' (Return e) = typeCheck' e
+typeCheck' (Assert c) = do
+  btype <- typeCheck' c
+  case btype of
+    BoolType -> return BoolType
+    t -> error $ "Assert condition must be BoolType, got: " ++ sep t btype
+typeCheck' (Equal a b) = do
+  atype <- typeCheck' a
+  btype <- typeCheck' b
+  if atype == btype
+    then return BoolType
+    else error $ "Equality test between values with different types: " ++ sep atype btype
```

Comment:
`typeCheck` currently uses `error` everywhere, should I change them all to `left`?